contracts/account/manager/src/contract.rs

```diff
@@ -35,11 +36,10 @@ pub(crate) const MAX_TITLE_LENGTH: usize = 64;
 #[cfg_attr(feature = "export", cosmwasm_std::entry_point)]
 pub fn migrate(deps: DepsMut, _env: Env, _msg: MigrateMsg) -> ManagerResult {
     let version: Version = CONTRACT_VERSION.parse().unwrap();
-    let storage_version: Version = get_contract_version(deps.storage)?.version.parse().unwrap();
-    if storage_version < version {
-        set_contract_version(deps.storage, MANAGER, CONTRACT_VERSION)?;
-        migrate_module_data(deps.storage, MANAGER, CONTRACT_VERSION, None::<String>)?;
-    }
+
+    assert_contract_upgrade(deps.storage, MANAGER, version)?;
+    set_contract_version(deps.storage, MANAGER, CONTRACT_VERSION)?;
+    migrate_module_data(deps.storage, MANAGER, CONTRACT_VERSION, None::<String>)?;
```

Comment:
Why are we adding module data to the manager, proxy, factories etc?
Imo module data is only useful in the modules (app,api) hence its name. 
it contains dependency information that is irrelevant to these contracts. The regular CW2 is good for these contracts. Less state = better

Reply:
Maybe we can have a separate `contract_metadata` or something?

Reply:
That's something we can introduce later. Also these contracts are used purely internally so who is going to value this data? We would also need to update the instantiation and possibly allow for updates, etc. 
This is ballooning complexity without any upside for imo, hence i'd like to keep it to the modules (where there is an actual upside to storing the schema metadata)

Reply:
If we decide to change our minds on the implementation of this data after we rolled it out we're going to have a lot of work on each of these contracts to reflect that as well.